hibernate-core/src/test/java/org/hibernate/orm/test/set/SetOperationTest.java

```diff
@@ -91,6 +92,7 @@ public void testUnionAllLimit(SessionFactoryScope scope) {
 
     @Test
     @RequiresDialectFeature(feature = DialectFeatureChecks.SupportsUnion.class)
+    @SkipForDialect(dialectClass = FirebirdDialect.class, reason = "Firebird does not support parenthesized select in UNION")
```

Comment:
We can fix that. I think some other databases are a bit picky there as well and require a select wrapper.

Reply:
In the `FirebirdSqlAstTranslator`, please add the following override:

```
	@Override
	protected boolean supportsSimpleQueryGrouping() {
		// Firebird is quite strict i.e. it requires `select .. union all select * from (select ...)`
		// rather than `select .. union all (select ...)`
		return false;
	}
```

Reply:
Thanks, I'll do that

Reply:
This also solved two failing tests in `InsertUpdateTests`